File: backend/Flowline.Domain/Entities/TimeEntry.cs
namespace Flowline.Domain.Entities;

public sealed class TimeEntry
{
    public required Guid Id { get; init; }
    public required Guid TaskId { get; init; }
    public required DateTime StartTime { get; init; }
    public DateTime? EndTime { get; init; } // null = still running
    public string? Notes { get; init; }
    public required DateTime CreatedAt { get; init; }

    // Navigation property
    public TaskEntity Task { get; set; } = null!;

    // Computed property
    public TimeSpan? Duration => EndTime.HasValue
        ? EndTime.Value - StartTime
        : DateTime.UtcNow - StartTime;
}

File: frontend/src/types/timeEntry.ts
import { Task } from './task';

export interface TimeEntry {
  id: string;
  taskId: string;
  startTime: string;
  endTime?: string;
  duration?: number; // in seconds
  notes?: string;
  createdAt: string;
  task?: Task;
}

export interface StartTimerRequest {
  taskId: string;
}

export interface TimerTickEvent {
  id: string;
  taskId: string;
  startTime: string;
  duration: number;
  elapsedTime: {
    days: number;
    hours: number;
    minutes: number;
    seconds: number;
  };
}
